src/covid_model_seiir_pipeline/pipeline/forecasting/model/ode_systems/variant.py

```diff
@@ -0,0 +1,581 @@
+                          susceptible, exposed, infectious1, infectious2, removed,
+                          susceptible_variant, exposed_variant,
+                          params, b_wild, b_variant,
+                          outflow_map):
+    # Normal epi
+    outflow_map[susceptible, exposed] += (b_wild + params[theta_plus]) * y[susceptible]
+    outflow_map[exposed, infectious1] += params[sigma] * y[exposed]
+    outflow_map[infectious1, infectious2] += params[gamma1] * y[infectious1]
+    outflow_map[infectious2, removed] += params[p_cross_immune] * params[gamma2] * y[infectious2]
+    # Theta minus
+    outflow_map[exposed, removed] += params[theta_minus] * y[exposed]
+    # Variant
+    outflow_map[susceptible, exposed_variant] += b_variant * y[susceptible]
+    outflow_map[infectious2, susceptible_variant] += (1 - params[p_cross_immune]) * params[gamma2] * y[infectious2]
+
+    return outflow_map
+
+
+@numba.njit
+def seiir_transition(y,
+                     susceptible, exposed, infectious1, infectious2, removed,
+                     params, beta_s,
+                     outflow_map):
+    outflow_map[susceptible, exposed] += beta_s * y[susceptible]
+    outflow_map[exposed, infectious1] += params[sigma] * y[exposed]
+    outflow_map[infectious1, infectious2] += params[gamma1] * y[infectious1]
+    outflow_map[infectious2, removed] += params[gamma2] * y[infectious2]
+
+    return outflow_map
+
+
+@numba.njit
+def get_vaccines_out(y, vaccines, params, b_wild, b_variant_s, b_variant_s_variant):
+    # Allocate our output space.
+    vaccines_out = np.zeros((y.size, vaccines.size))
+    v_total = vaccines.sum()
+    # Don't vaccinate if no vaccines to deliver.
+    if not v_total:
+        return vaccines_out
+
+    n_unvaccinated = count_unvaccinated(y)
+    if n_unvaccinated:
+        # S has many kinds of effective and ineffective vaccines
+        vaccines_out = vaccinate_from_s(
+            y, vaccines,
+            params, b_wild, b_variant_s,
+            n_unvaccinated, v_total,
+            vaccines_out
+        )
+        # S_variant has many kinds of effective and ineffective vaccines
+        vaccines_out = vaccinate_from_s_variant(
+            y, vaccines,
+            b_variant_s_variant,
+            n_unvaccinated, v_total,
+            vaccines_out
+        )
+        # Folks in E, I1, I2, R only unprotected.
+        vaccines_out = get_unprotected_vaccines_from_not_s(
+            y, vaccines_out, params, n_unvaccinated, v_total,
+            e, i1, i2, r,
+        )
+        vaccines_out = get_unprotected_vaccines_from_not_s(
+            y, vaccines_out, params, n_unvaccinated, v_total,
+            e_variant, i1_variant, i2_variant, r_variant,
+        )
+
+    v_total_empirical = vaccines_out.sum()
+#    if v_total:
+#       percent_loss = abs(v_total_empirical - v_total) / v_total  * 100
+#       if percent_loss > .05:
+#           print('Losing more than 5% vaccines. Percent loss: ', percent_loss)
```

Comment:
Is this dead code useful moving forward or should it be shed meow?

Reply:
Still useful. It's the only place I have this system written down.

Reply:
Which is a bad use of code space, but I'm pressed for time.